xarray/core/dataset.py

```diff
@@ -332,17 +332,24 @@ def _initialize_curvefit_params(params, p0, bounds, func_args):
     """Set initial guess and bounds for curvefit.
     Priority: 1) passed args 2) func signature 3) scipy defaults
     """
+    from xarray.core.computation import where
 
     def _initialize_feasible(lb, ub):
         # Mimics functionality of scipy.optimize.minpack._initialize_feasible
         lb_finite = np.isfinite(lb)
         ub_finite = np.isfinite(ub)
-        p0 = np.nansum(
-            [
-                0.5 * (lb + ub) * int(lb_finite & ub_finite),
-                (lb + 1) * int(lb_finite & ~ub_finite),
-                (ub - 1) * int(~lb_finite & ub_finite),
-            ]
+        p0 = where(
```

Comment:
An alternative I considered here (see cd685ba0) is
```python
 p0 = np.nansum(
	 [
		 0.5 * (lb + ub) * int(lb_finite & ub_finite),
		 (lb + 1) * int(lb_finite & ~ub_finite),
		 (ub - 1) * int(~lb_finite & ub_finite),
	 ]
		 0.5 * (lb + ub) * (lb_finite & ub_finite),
		 (lb + 1) * (lb_finite & ~ub_finite),
		 (ub - 1) * (~lb_finite & ub_finite),
	 ],
	 axis=0,
 )
```

which is closer to the original, but it gave warnings about the multiplication (the original version also issues warnings if you remove the `int()` cast). Not sure which is clearer.